### ISSUE TITLE
Add OpenID Connect support

### ISSUE THREAD

**Issue by burnoutberni** [Budibase/budibase#1806]
Hi. I recently stumbled across BudiBase and I'm pretty excited about the opportunities for the non-profit where I'm volunteering on the tech team.

We use [Keycloak](https://www.keycloak.org/) for our identity management, so every volunteer has a Single Sign-On account. For us to productively use BudiBase, we'd need to integrate it, but unfortunately BudiBase doesn't support SAML or OIDC – the two standards supported by Keycloak. I saw a [thread on Reddit](https://www.reddit.com/r/opensource/comments/l6yyvr/budibase_build_modern_web_apps_on_your_own/gldkqwu/), where this was mentioned, and someone on your team said, it was an important concern. After looking around here, it doesn't seem like anyone is working on it, which is why I'm opening this issue.

It seems to me like it should be rather easy to implement Open ID Connect (OIDC) support, since you already use OAuth2 for identity management with Google. OIDC is a simple layer on top of OAuth2 to allow for arbitrary Identity Providers (which could be Google, Github or any other OAuth2 provider, but also a privately hosted IdP). I also feel like it would be very useful for many organizations, not just mine, so I really hope you'd consider adding OIDC sometimes soon.

Thanks in any case for your hard work!

**Comment by shogunpurple** [comment]
Hi @burnoutberni - thanks for raising this. This is something that has been raised a bit in our github discussions community:

https://github.com/Budibase/budibase/discussions/1630

We will be looking at Active Directory Authentication over the next few weeks, but with a view to expand into other auth providers as well.

```
It seems to me like it should be rather easy to implement Open ID Connect (OIDC) support, since you already use OAuth2 for identity management with Google.
```

As an open source project, we are always open to contributions - so if you have knowledge around passportJS (the middleware our auth is built on) and openID connect, it would be good to have your insight here around integrating the two. 

We will keep the community up to date as we work on other budibase authentication methods - thanks for raising this!

**Comment by burnoutberni** [comment]
Thanks for your swift reply, @shogunpurple. I didn't see that discussion before, so thanks also for pointing that out.

Unfortunately, I do not have hands-on experience with OIDC implementation, I've only ever done minor tweaks to software in case things didn't work out smoothly.

PassportJS does look pretty straight-forward though and since any OIDC implementation will be very similar to Google OAuth2, I think the heavy lifting is already done. If I get around to it, I might give it a try, but no promises.

**Comment by Rory-Powell** [comment]
OIDC support has been shipped in version `0.9.83`
Check out the documentation for more details: https://docs.budibase.com/admin/authentication
